cli/src/commands/config/set.rs

```diff
@@ -46,7 +56,63 @@ pub fn cmd_config_set(
             path = config_path.display()
         )));
     }
+
     // TODO(#531): Infer types based on schema (w/ --type arg to override).
     let value = parse_toml_value_or_bare_string(&args.value);
+
+    // If the user is trying to change the author config, we should warn them that
+    // it won't affect the working copy author
+    if args.name == ConfigNamePathBuf::from_iter(vec!["user", "name"]) {
+        check_wc_author(ui, command, &value, AuthorChange::Name)?;
+    } else if args.name == ConfigNamePathBuf::from_iter(vec!["user", "email"]) {
+        check_wc_author(ui, command, &value, AuthorChange::Email)?;
+    };
+
     write_config_value_to_file(&args.name, value, &config_path)
 }
+
+/// Returns the commit of the working copy if it exists.
+fn maybe_wc_commit(helper: &WorkspaceCommandHelper) -> Option<Commit> {
+    let repo = helper.repo();
+    let maybe_wc_commit = helper
+        .get_wc_commit_id()
+        .map(|id| repo.store().get_commit(id))
+        .transpose()
+        .unwrap();
```

Comment:
Oh, there's a real error which I didn't notice, sorry. Suppose it's okay to suppress warning on storage error, it can be turned into `None`.
```rust
    let id = helper.get_wc_commit_id()?;
    repo.store().get_commit(id).ok()
```